### PR TITLE
Local build bugs - ncurses and libuv

### PR DESCRIPTION
**Pull request by max-allan**
The checksum for libuv was not the same as what is currently downloading. Fixed that.
```
$ curl  https://dist.libuv.org/dist/v1.47.0/libuv-v1.47.0-dist.tar.gz | sha256sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1615k  100 1615k    0     0  2815k      0 --:--:-- --:--:-- --:--:-- 2810k
72a187104662b47f2a2b204da39d2acb05cf22a4fcb13ceaebe3b0ed0c0e2e43  -
```
The GPG signature for the file uses a key mentioned in the project's MAINTAINERS file.

### CODE REVIEW

File: ncurses.yaml [hunk 1]
```diff
@@ -17,19 +17,10 @@ environment:
...
-    replace: -$1
-    to: mangled-package-version
-
 pipeline:
   - uses: fetch
     with:
-      uri: https://invisible-mirror.net/archives/ncurses/current/ncurses-${{vars.mangled-package-version}}.tgz
-      expected-sha512: 262e1efbb5fbfe44f8efe2737c08cdf304f39671bab03cdc1c61ddf2ff7025e3f335d7b06ff812344b9d305c6c7baac7d624a691011caea6ab39932f417ec58e
-
+https://github.com/max-allan/os
```
Comment (rawlingsj) [inline ncurses.yaml:23]: I think this change is accidental?

**Review by rawlingsj — DISMISSED** [review]
Was this fixing a bug for the ncurses build?  I'm not sure that the ncurses changes are needed.

File: ncurses.yaml (hunk 1 above)
Comment (max-allan) [reply ncurses.yaml]: Whoops!

**Comment by max-allan** [comment]
@rawlingsj the original source of ncurses is moved/gone.
```
curl https://invisible-mirror.net/archives/ncurses/current/ncurses-6.4.tgz
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL was not found on this server.</p>
<p>Additionally, a 404 Not Found
error was encountered while trying to use an ErrorDocument to handle the request.</p>
</body></html>
```

So

File: ncurses.yaml [hunk 2]
```diff
@@ -17,18 +17,11 @@ environment:
...
-    match: _p(\d+)$
-    replace: -$1
-    to: mangled-package-version
-
 pipeline:
   - uses: fetch
     with:
-      uri: https://invisible-mirror.net/archives/ncurses/current/ncurses-${{vars.mangled-package-version}}.tgz
-      expected-sha512: 262e1efbb5fbfe44f8efe2737c08cdf304f39671bab03cdc1c61ddf2ff7025e3f335d7b06ff812344b9d305c6c7baac7d624a691011caea6ab39932f417ec58e
+      uri: https://distfiles.alpinelinux.org/distfiles/edge/ncurses-${{package.version}}-20230715.tgz
```
Comment (vaikas) [inline ncurses.yaml:23]: Looking at the dates, isn't this pulling in an older version now?

**Comment by rawlingsj** [comment]
@max-allan thanks for that, as it happens, it will have caused apk resolver issues if we moved away from using a date in the `package.version`.

We've since merged https://github.com/wolfi-dev/os/commit/405cb361c57e0f0194c4ef9a38656e64d092c892.

Would you mind if we remove the ncurses change from this PR and we can get the libuv change merged?

Thanks for spotting and helping with this!

**Comment by max-allan** [comment]
@rawlingsj , sorry for the delay. Real life got in the way. Change is only libuv.yaml now.

**Review by kaniini — APPROVED** [review]

**Pull request merged**